src/main/src/main/java/org/geoserver/security/SecuredResourceNameChangeListener.java

```diff
@@ -51,37 +54,32 @@ public void handleAddEvent(CatalogAddEvent event) throws CatalogException {
 
     @Override
     public void handleRemoveEvent(CatalogRemoveEvent event) throws CatalogException {
-        String resourceWorkSpaceName = null;
-        CatalogInfo resourceInfo = event.getSource();
-        String resourceName = null;
-        if (resourceInfo instanceof LayerInfo) {
-            resourceWorkSpaceName =
-                    ((LayerInfo) resourceInfo).getResource().getStore().getWorkspace().getName();
-            resourceName = ((LayerInfo) resourceInfo).getName();
-        } else if (event.getSource() instanceof LayerGroupInfo) {
-            resourceWorkSpaceName = ((LayerGroupInfo) resourceInfo).getWorkspace().getName();
-            resourceName = ((LayerGroupInfo) resourceInfo).getName();
-        }
-
-        if (resourceWorkSpaceName != null && resourceInfo != null) {
-            // if a security rule exists for the layer whose name has been changed
-            if (layerHasSecurityRule(resourceWorkSpaceName, resourceName)) {
-                LOGGER.info(
-                        String.format(
-                                "Removing Security Rules for removed layer: %s", resourceName));
-
-                List<DataAccessRule> rulesToRemove = getDataAccessRule(resourceInfo, resourceName);
-                for (DataAccessRule r : rulesToRemove) {
-                    dao.removeRule(r);
-                }
-                try {
-                    dao.storeRules();
-                } catch (IOException e) {
-                    LOGGER.log(Level.SEVERE, e.getMessage(), e);
-                    throw new CatalogException(e);
-                }
+        Predicate<DataAccessRule> filter;
+        final String layerName;
+        final CatalogInfo eventSource = event.getSource();
+        if (eventSource instanceof LayerInfo) {
+            LayerInfo l = (LayerInfo) eventSource;
+            WorkspaceInfo ws = l.getResource().getStore().getWorkspace();
+            layerName = l.getName();
+            filter = filter(ws.getName()::equalsIgnoreCase, layerName::equalsIgnoreCase);
```

Comment:
I see you're just replicating the original behavior but... wondering why it's case insensitive.
Layer names are case sensitive in GeoServer, and so seems the application of rules in DefaultResourceAccessManager.

Checked if there was any discussion about it in the original PR, but [there seems to have been none](https://github.com/geoserver/geoserver/pull/3644).

Reply:
I don't know, just wouldn't dare changing the current behavior